persist/provider/DashViewProvider.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+
+import {PersistenceProvider} from './PersistenceProvider';
+import {throwIf} from '../../utils/js';
+
+/**
+ * PersistenceProvider that stores state within a DashView.
+ */
+export class DashViewProvider extends PersistenceProvider {
```

Comment:
Minor pedantry (and *not* suggesting we change anything), but writing / reading to DashView state isn't *really* persistence... :) don't have any better suggestions.